<launch>

	<!-- Machine -->
	<machine name="octohost" address="localhost" env-loader="/opt/ros/indigo/env.sh"/>
	<arg name="machine" default="octohost" />

	<!-- Octomap default values of parameters --> 
	<arg name="resolution" default="0.02"/>
	<arg name="max_range" default="1.5"/>
	<arg name="cloud_in" default="/asus/depth_registered/points"/>

	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server" machine="$(arg machine)">
		<param name="resolution" value="$(arg resolution)" />
		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<param name="frame_id" type="string" value="world" />
		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model/max_range" value="$(arg max_range)" />
		<!-- For maximum performance when building a map, set to false -->
		<param name="latch" value="false" />
		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="$(arg cloud_in)" />
	</node>

</launch>
